staging/src/k8s.io/apiserver/pkg/cel/library/network_addrs.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"net/netip"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+// NetworkAddr provides a CEL function library extension of network address parsing functions.
```

Comment:
I would just call this "CIDR". It matches `net.ParseCIDR` and I'm pretty sure it's the term we most commonly use. Admittedly, it's not super accurate: "CIDR" ("Classless Inter-Domain Routing") technically refers to the entire concept of breaking up IPv4 subnets on arbitrary boundaries, and the syntax is more properly "CIDR notation", except also, CIDR notation can be used either to refer to subnets/masks ("`1.2.3.0/24`") or individual IPs ("`1.2.3.4/24`" meaning "the IP `1.2.3.4` and by the way it's on a `/24` subnet"), but we only want this to mean the former. But I don't think there's really any standard simple name for that...

(However, "network address" feels definitely bad to me, because `netlink.Addr` is used in the CNI code to mean the "`1.2.3.4/24`" case.)